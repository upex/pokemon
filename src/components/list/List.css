.pokemon-container {
  padding: 100px 10px 60px 10px;
  display: flex;
  align-items: center;
  justify-content: center;
}