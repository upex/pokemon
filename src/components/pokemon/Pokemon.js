import React from 'react';
import './Pokemon.css';

function Pokemon(props) {
  let { pokemon, id, url } = props;
  switch (pokemon) {
    case "nidoran-f":
      pokemon = "nidoranf";
      break;
    case "nidoran-m":
      pokemon = "nidoranm";
      break;
    case "deoxys-normal":
      pokemon = "deoxys";
      break;
    default:
      break;
  }
  const getId = (url) => {
    const id = url.substring(34, url.length - 1);
    return id;
  }
  return (
    <div className="pokemon-card">
      <img
        className="sprites"
        src={`http://pokestadium.com/sprites/xy/${pokemon}.gif`}
        alt={pokemon}
        id={id}
      />
      <div className="align-bottom">{pokemon} ({getId(url)})</div>
    </div>
  );
}

export default Pokemon;