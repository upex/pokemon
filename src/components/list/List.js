import React from 'react';
import PropTypes from 'prop-types';
import './List.css';
import Pokemon from '../pokemon/Pokemon';

function List(props) {
  let { items } = props;
  return (
    <div className="pokemon-container">
      {items.map((monster, index) =>
        <Pokemon key={monster.name} id={index + 1} url={monster.url} pokemon={monster.name} />
      )}
    </div>
  );
}
List.propTypes = {
  items: PropTypes.array,
};
export default List;