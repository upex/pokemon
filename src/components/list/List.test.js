import React from 'react';
import { shallow } from '../../enzyme';
import List from './List';
import Pokemon from '../pokemon/Pokemon';

describe('List tests', () => {
  it('renders pokemon-items', () => {
    const items = [
      {
        name:"bulbasaur",
        url:"https://pokeapi.co/api/v2/pokemon/1/"
      },
      {
        name:"ivysaur",
        url:"https://pokeapi.co/api/v2/pokemon/2/"
      },
      {
        name:"venusaur",
        url:"https://pokeapi.co/api/v2/pokemon/3/"
      }
    ];
    const wrapper = shallow(<List items={items} />);
    console.log(wrapper.debug());
    expect(wrapper.find('.pokemon-container')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a pokemon', () => {
    const items = [
      {
        name:"bulbasaur",
        url:"https://pokeapi.co/api/v2/pokemon/1/"
      },
      {
        name:"ivysaur",
        url:"https://pokeapi.co/api/v2/pokemon/2/"
      },
      {
        name:"venusaur",
        url:"https://pokeapi.co/api/v2/pokemon/3/"
      }
    ];
    const wrapper = shallow(<List items={items} />);
    // Check if an element in the Component exists
    expect(wrapper.find('.pokemon-card')).toBeDefined();
  });
});