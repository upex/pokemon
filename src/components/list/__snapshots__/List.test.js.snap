// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests renders a pokemon 1`] = `ShallowWrapper {}`;

exports[`List tests renders pokemon-items 1`] = `ShallowWrapper {}`;
